From d7fde9c83441b060309913a2fa61521fdbfe40c0 Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Mon, 18 Aug 2014 19:58:46 -0400
Subject: [block] nvme: Make admin timeout a module parameter

Message-id: <1408391935-24886-14-git-send-email-dmilburn@redhat.com>
Patchwork-id: 87897
O-Subject: [RHEL7.1 PATCH BZ 1111259 13/22] NVMe: Make admin timeout a module parameter
Bugzilla: 1111259
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Keith Busch <keith.busch@intel.com>

Signed-off-by: Keith Busch <keith.busch@intel.com>
[made admin_timeout static]
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>

(cherry picked from commit 9d43cf646eab948db81913379dacb1dcecacb1eb)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/nvme-core.c b/drivers/block/nvme-core.c
index e16682d..d59adf4 100644
--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -44,11 +44,15 @@
 
 #include <trace/events/block.h>
 
-#define NVME_Q_DEPTH 1024
+#define NVME_Q_DEPTH  1024
 #define SQ_SIZE(depth)  (depth * sizeof(struct nvme_command))
 #define CQ_SIZE(depth)  (depth * sizeof(struct nvme_completion))
-#define ADMIN_TIMEOUT (60 * HZ)
-#define IOD_TIMEOUT (retry_time * HZ)
+#define ADMIN_TIMEOUT  (admin_timeout * HZ)
+#define IOD_TIMEOUT  (retry_time * HZ)
+
+static unsigned char admin_timeout = 60;
+module_param(admin_timeout, byte, 0644);
+MODULE_PARM_DESC(admin_timeout, "timeout in seconds for admin commands");
 
 unsigned char io_timeout = 30;
 module_param(io_timeout, byte, 0644);
-- 
1.7.1